File: zaruba/servers.py
import os
import re


def _ids(value):
    return [item.strip() for item in re.split(r"[,\s]+", str(value or "")) if item.strip()]


def make_server(server_id):
    key = str(server_id)
    name = os.getenv(f"SERVER_{key}_NAME") or f"СЕРВЕР {key}"
    return {
        "id": key,
        "name": name,
        "query": os.getenv(f"SERVER_{key}_QUERY") or name,
        "game_id": os.getenv(f"SERVER_{key}_GAME_ID") or "",
        "coming_soon": os.getenv(f"SERVER_{key}_SOON") == "1",
        "bot_token": os.getenv(f"SERVER_{key}_BOT_TOKEN") or "",
        "addr": os.getenv(f"SERVER_{key}_ADDR") or "",
        "rcon_host": os.getenv(f"SERVER_{key}_RCON_HOST") or "",
        "rcon_port": os.getenv(f"SERVER_{key}_RCON_PORT") or "",
        "rcon_password": os.getenv(f"SERVER_{key}_RCON_PASSWORD") or "",
        "seeds": _ids(os.getenv(f"SERVER_{key}_SEEDS")),
    }


SERVERS = [make_server(i) for i in (1, 2, 3, 4)]


def visible_servers():
    return [
        server
        for server in SERVERS
        if server["rcon_host"] or server["addr"] or server["coming_soon"]
    ]


def get_server(server_id):
    key = re.sub(r"^(server|сервер)[-_\s]*", "", str(server_id or "").strip().lower())
    return next((server for server in SERVERS if server["id"] == key), None)

File: src/servers.js
function parseIds(value) {
  return String(value || "")
    .split(/[,\s]+/)
    .map((id) => id.trim())
    .filter(Boolean);
}

function makeServer(id) {
  const key = String(id);
  return {
    id: key,
    name: process.env[`SERVER_${key}_NAME`] || `СЕРВЕР ${key}`,
    query:
      process.env[`SERVER_${key}_QUERY`] ||
      process.env[`SERVER_${key}_NAME`] ||
      `СЕРВЕР ${key}`,
    gameId: process.env[`SERVER_${key}_GAME_ID`] || "",
    comingSoon: process.env[`SERVER_${key}_SOON`] === "1",
    botToken: process.env[`SERVER_${key}_BOT_TOKEN`] || "",
    addr: process.env[`SERVER_${key}_ADDR`] || "",
    rconHost: process.env[`SERVER_${key}_RCON_HOST`] || "",
    rconPort: process.env[`SERVER_${key}_RCON_PORT`] || "",
    rconPassword: process.env[`SERVER_${key}_RCON_PASSWORD`] || "",
    seeds: parseIds(process.env[`SERVER_${key}_SEEDS`]),
  };
}

export const SERVERS = [1, 2, 3, 4].map(makeServer);

export function visibleServers() {
  return SERVERS.filter(
    (server) => server.rconHost || server.addr || server.comingSoon
  );
}

export function getServer(id) {
  const key = String(id || "")
    .trim()
    .toLowerCase()
    .replace(/^server[-_\s]*/, "")
    .replace(/^сервер[-_\s]*/, "");
  return SERVERS.find((server) => server.id === key) || null;
}

export function serverChoices() {
  return visibleServers().map((server) => ({ name: server.name, value: server.id }));
}
